servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -149,27 +152,28 @@ public String getServiceName() {
 
   @Override
   public void processJournalEntry(JournalEntry entry) throws IOException {
-    if (entry instanceof InodeEntry) {
-      mInodeTree.addInodeFromJournal((InodeEntry) entry);
-    } else if (entry instanceof InodeLastModificationTimeEntry) {
-      InodeLastModificationTimeEntry modTimeEntry = (InodeLastModificationTimeEntry) entry;
+    Message innerEntry = JournalProtoUtils.unwrap(entry);
```

Comment:
`instanceof` is cheap, and we don't often need to recover, so I think it makes sense to keep this code as simple as possible. If it ever becomes a perf issue though a registry could give us constant time.